code/area.dm

```diff
@@ -3544,6 +3544,30 @@ ABSTRACT_TYPE(/area/station/ai_monitored/storage/)
 	teleport_blocked = 1
 	spy_secure_area = TRUE
 	station_map_colour = MAPC_ARMOURY
+	var/static/list/entered_ckeys = list()
+
+	Entered(atom/movable/A, atom/oldloc)
+		. = ..()
+		if (current_state < GAME_STATE_FINISHED)
+			if(istype(A, /mob/living))
+				var/mob/living/M = A
+				if(!M.client)
+					return
+				if(!M.client.holder)
+					return
+				if(M.client.ckey in entered_ckeys)
+					return
+				if (M.mind && M.mind.assigned_role == "Head of Security")
+					return
+				var/armory_auth = FALSE
+				for_by_tcl(O, /obj/machinery/computer/riotgear)
+					if (O.authed)
+						armory_auth = TRUE
+						break
+				entered_ckeys += M.client.ckey
+				SPAWN(120 SECONDS)
+					entered_ckeys -= entered_ckeys[1]
```

Comment:
probably better to remove the specific ckey rather than just the first one, could cause jank?

Reply:
I understood the order wouldn't change?